File: app/src/main/java/c/m/storyapp/authentication_check/domain/use_case/logout_use_case/LogoutUseCase.kt
package c.m.storyapp.authentication_check.domain.use_case.logout_use_case

import c.m.storyapp.authentication_check.domain.repository.AuthenticationCheckRepository
import c.m.storyapp.common.util.Constants
import c.m.storyapp.common.util.SimpleResource
import kotlinx.coroutines.flow.Flow
import javax.inject.Inject

class LogoutUseCase @Inject constructor(private val authenticationCheckRepository: AuthenticationCheckRepository) {
    suspend operator fun invoke(): Flow<SimpleResource> {
        return authenticationCheckRepository.saveToken(Constants.BLANK_STRING)
    }
}

File: lib/authentication_check/domain/use_case/logout_use_case/logout_use_case.dart
import 'package:injectable/injectable.dart';
import 'package:story_app/authentication_check/domain/repository/authentication_check_repository.dart';
import 'package:story_app/common/util/constants.dart';

@injectable
class LogoutUseCase {
  final AuthenticationCheckRepository _authenticationCheckRepository;

  LogoutUseCase(this._authenticationCheckRepository);

  Future<void> call() async {
    await _authenticationCheckRepository.saveToken(Constants.blankString);
  }
}
